packages/pwa-buildpack/lib/BuildBus/declare-base.js

```diff
@@ -0,0 +1,6 @@
+const { SyncHook } = require('tapable');
+
+module.exports = api => {
+    api.declareTarget('webpackCompiler', new SyncHook(['compiler']));
+    api.declareTarget('specialFeatures', new SyncHook(['featuresByModule']));
```

Comment:
> BuildBus has some builtin targets
> specialFeatures: Runs before the compilation begins...
> webpackCompiler: Runs when a Webpack compiler is created....

What would you think of naming these things for when they run, like event handlers? We could export targets for use like `onCompilationStart`, `onCompilationEnd`, `onCompilerCreated`, etc.

Reply:
@supernova-at I love this suggestion! Let's make our API super obvious of when things run :)

Reply:
I'd be more than happy to do that for our own targets. Webpack's internal hooks do a more `beforeThing` / `afterThing` model; what do you think of that?

Reply:
@zetlen I'm happy with either, as long as it's reasonably clear from the event name when the target will be executed I think we're cool. if webpack is doing before / after maybe that makes the most sense as this is a webpack-y type thing-y

Reply:
`upon` it is, then!